modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -120,14 +126,96 @@ public Channel getChannel() {
         return channel;
     }
 
-    @Override
-    public String header(String name) {
-        return request.headers().get(name);
-    }
+    /**
+     * A wrapper of {@link HttpHeaders} that implements a map to prevent copying unnecessarily. This class does not support modifications
+     * and due to the underlying implementation, it performs case insensitive lookups of key to values.
```

Comment:
The headers were case-insensitive before. Both the Netty 3 and 4 implementations implemented this.